packages/app/src/specs/SpecsListRunWatcher.vue

```diff
@@ -0,0 +1,50 @@
+/**
+ Non rendering component used to watch running runs and emit an event
+ when a run has changed. Used by the specs list to update spec information
+ during a RUNNING run.
+*/
+<script setup lang="ts">
+import { gql, useSubscription } from '@urql/vue'
+import { computed } from 'vue'
+import { SpecsListRunWatcherDocument } from '../generated/graphql'
+import type { RelevantRunInfo } from '@packages/frontend-shared/cypress/support/generated/test-graphql-types.gen'
+
+/**
+ * Subscription to watch a run.
+ * Values being queried are there to tell the polling source
+ * which fields to watch for changes
+ */
+gql`
+subscription SpecsListRunWatcher($id: ID!) {
+  relevantRunSpecChange(runId: $id) {
+    id
+    status
+    totalInstanceCount
+    completedInstanceCount
+  }
+}
```

Comment:
I think this pattern makes sense - each run is responsible for setting up its own subscription and getting the data, as well as cleanup.